Mathlib/Algebra/Divisibility/Ring.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.Algebra.Divisibility.Basic
+import Mathlib.Data.Nat.Choose.Sum
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Tactic.Linarith
+
+/-!
+# Lemmas about divisibility in rings
+-/
+
+variable {R : Type*} {x y : R}
+
+lemma dvd_nsmul_of_dvd [Semiring R] {n : ℕ} (h : x ∣ y) : x ∣ n • y :=
+  ⟨n • Classical.choose h, by rw [mul_smul_comm, ← Classical.choose_spec h]⟩
+
+namespace Commute
```

Comment:
Do you think it's worth repeating these results for `CommSemiring`, like we do for lots of other `Commute` results?

Reply:
I'm happy to do this if you like but I tend to think this pattern of repeating results is only worth it for very broadly applicable results and the lemmas proposed here are quite specialised. Would you still like me to do it?

Reply:
No, it's fine without; I just wanted to check you had decided it was unhelpful.